### PR TITLE
added login component

### PR DESCRIPTION
**Pull request by Yh010**
Component created. Working on re routing and storing tokens in local storage

### CODE REVIEW

**Comment by Yh010** [comment]
@aadeshkulkarni we would also need a logout button right

**Comment by aadeshkulkarni** [comment]
> @aadeshkulkarni we would also need a logout button right

Yes

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+.history
```
Comment (aadeshkulkarni) [inline .gitignore]: Git should ignore all files & folders inside .history 
change .history to .history/

File: src/Components/AuthContext.js
Comment (aadeshkulkarni) [inline src/Components/AuthContext.js:1]: Remove empty files

File: src/App.js [hunk 2]
```diff
@@ -18,11 +18,11 @@ const router = createBrowserRouter([{
   element: <AppLayout />,
   children: [
     {
-      path: "/",
+      path: "/main",
```
Comment (aadeshkulkarni) [inline src/App.js]: The default url "/" should point to Main component. If Auth is not present in the Main component, it should redirect to "/login" route.

File: src/Components/Main.js [hunk 3]
```diff
@@ -29,7 +28,8 @@ const Main = () => {
 
   };
   return (
-    <div className="flex justify-center items-center mt-12">
+    <div>
```
Comment (aadeshkulkarni) [inline src/Components/Main.js:31]: Any particular reason why an extra div is added here ?

**Review by aadeshkulkarni — CHANGES_REQUESTED** [review]

File: src/Components/Main.js [hunk 4]
```diff
@@ -55,6 +55,11 @@ const Main = () => {
         </div>
       </form>
     </div>
+
+    </div>
+
+
```
Comment (aadeshkulkarni) [inline src/Components/Main.js]: remove unnecessary whitespaces

File: src/Components/Main.js (hunk 3 above)
Comment (Yh010) [reply src/Components/Main.js:31]: Was thinking of adding a login/logout/signup button here. when the user logs in,only then he will be able to see his projects, else he'll just see the contents of the page or maybe some default page stating "signup to see your contents".

**Comment by Yh010** [comment]
@aadeshkulkarni what if we add a default landing page, which contains a "get started " button, which when clicked will redirect to the login process, and upon successful login, the user is redirected to his dashboard?

Something of this sort: https://rxresu.me/

**Comment by aadeshkulkarni** [comment]
> @aadeshkulkarni what if we add a default landing page, which contains a "get started " button, which when clicked will redirect to the login process, and upon successful login, the user is redirected to his dashboard?
> 
> Something of this sort: https://rxresu.me/

That's the landing page. It's development is in the pipeline for future. For now, let's keep the default directory as the Main component.

**Comment by aadeshkulkarni** [comment]
Closing this PR since the feature is fixed by @Priyanga-Developer

**Pull request closed without merging**